qiskit_alice_bob_provider/plugins/state_preparation.py

```diff
@@ -416,10 +372,25 @@ def pass_manager(
                 basis_translator_target,
             ),
         )
-        if need_synthesis:
-            custom_pm.append(_get_unitary_synthesis(pass_manager_config))
+        if use_solovay_kitaev and need_second_sk and solovay_kitaev:
+            custom_pm.append(solovay_kitaev)
 
         custom_pm += generate_embed_passmanager(
             pass_manager_config.coupling_map
         )
         return custom_pm
+
+    def pass_manager_with_sk(
```

Comment:
Out of curiosity, why defining a new method pass_manager_with_sk (compared to simply having the pass_manager) ?